tiny_dnn/layers/layer.h

```diff
@@ -530,16 +530,16 @@ class layer : public node {
 
     void update_weight(optimizer *o, serial_size_t batch_size) {
         float_t rcp_batch_size = float_t(1) / float_t(batch_size);
+        vec_t diff;
         for (serial_size_t i = 0; i < static_cast<serial_size_t>(in_type_.size()); i++) {
             if (trainable() && is_trainable_weight(in_type_[i])) {
-                vec_t diff;
                 vec_t& target = *get_weight_data(i);
-
                 ith_in_node(i)->merge_grads(&diff);
                 std::transform(diff.begin(), diff.end(),
                                diff.begin(), [&](float_t x) { // NOLINT
                                   return x * rcp_batch_size; });
-                o->update(diff, target);
+                bool parallelize = (target.size() >= 512);
```

Comment:
I just added below comment in recent commit.
```
// parallelize only when target size is big enough to mitigate thread spawning overhead
bool parallelize = (target.size() >= 512);
o->update(diff, target, parallelize);
```
I added `bool parallelize` parameter to `optimizer::update` method and its descendants.
The newly added parameter is used for 1st parameter of `for_i` function (that has triple argument prototype).  `for_i` function can selectively create multi-threads but doing so is not favourable when size of data is small.